### PR TITLE
ir: Don't parse non-semantic-children cursor as inner structs.

### PR DESCRIPTION
**Pull request by emilio**
r? @fitzgen

### CODE REVIEW

File: src/ir/comp.rs [hunk 1]
```diff
@@ -605,11 +605,17 @@ impl CompInfo {
                 CXCursor_UnionDecl |
                 CXCursor_ClassTemplate |
                 CXCursor_ClassDecl => {
+                    // We can find non-semantic children here, see:
+                    // https://github.com/servo/rust-bindgen/issues/482
```
Comment (fitzgen) [inline src/ir/comp.rs]: Please leave a one or two sentence description of why/when we can find non-semantic children here, and only link the issue if you think it is still relevant with those sentences.

File: src/ir/comp.rs [hunk 2]
```diff
@@ -605,11 +605,17 @@ impl CompInfo {
...
+                    if cur.semantic_parent() != cursor {
+                        return CXChildVisit_Continue;
```
Comment (fitzgen) [inline src/ir/comp.rs]: Is it worth lifting this check and early return up out to right before the match?

**Comment by fitzgen** [comment]
(r=me, but I will let you delegate to bors so you can decide how to respond to my review feedback)

**Comment by emilio** [comment]
@bors-servo r=fitzgen

 * Improved the comment.
 * I don't think the check is worth uplifting

Thanks for the review Nick! :)

**Comment by emilio** [comment]
Fixed a few places where we were passing `parent_id` down incorrectly.

@bors-servo r=fitzgen

**Pull request merged**